vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/AdamDeviceProvider.kt

```diff
@@ -127,7 +135,9 @@ class AdamDeviceProvider(
     override suspend fun terminate() {
         bootWaitContext.close()
         channel.close()
-        deviceEventsChannel.cancel()
+        synchronized(deviceEventsChannelLock) {
```

Comment:
Can we use https://kotlinlang.org/docs/reference/coroutines/shared-mutable-state-and-concurrency.html#mutual-exclusion
instead of synchronized?